bindings/python/cstest_py/src/cstest_py/cstest.py

```diff
@@ -293,27 +285,17 @@ def test(self) -> TestResult:
         if self.skip:
             log.info(f"Skip {self}\nReason: {self.skip_reason}")
             return TestResult.SKIPPED
-
-        try:
-            self.input.setup()
-        except Exception as e:
-            log.error(f"Setup failed at with: {e}")
-            traceback.print_exc()
+        
+        def _log_err(step: str, e: Exception):
+            log.error(e)
             return TestResult.ERROR
 
         try:
+            self.input.setup()
             insns = self.input.decode()
-        except Exception as e:
-            log.error(f"Decode failed with: {e}")
-            traceback.print_exc()
-            return TestResult.ERROR
-
-        try:
             return self.expected.compare(insns, self.input.arch_bits)
         except Exception as e:
-            log.error(f"Compare expected failed with: {e}")
-            traceback.print_exc()
-            return TestResult.ERROR
+            return _log_err("Setup", e)
```

Comment:
Please don't merge those steps.

These are the bindings, so we need to distinguish if the setup, decode, or comparing step failed.